aspnetcore/security/authentication/azure-ad-b2c-webapi.md

```diff
@@ -0,0 +1,208 @@
+---
+title: Cloud authentication in web APIs with Azure Active Directory B2C
+author: camsoper
+description: Discover how to set up Azure Active Directory B2C authentication with ASP.NET Core Web API. Test the authenticated web API with Postman.
+ms.author: casoper
+manager: wpickett
+ms.date: 01/25/2018
+ms.topic: tutorial
+ms.technology: aspnet
+ms.prod: asp.net-core
+uid: security/authentication/azure-ad-b2c-webapi
+custom: mvc
+---
+# Cloud authentication in web APIs with Azure Active Directory B2C
+
+By [Cam Soper](https://twitter.com/camsoper)
+
+[Azure Active Directory B2C](/azure/active-directory-b2c/active-directory-b2c-overview) (Azure AD B2C) is a cloud identity management solution for your web and mobile apps. The service provides authentication for apps hosted in the cloud and on-premises. Authentication types include include individual accounts, social network accounts, and federated enterprise accounts.  Additionally, Azure AD B2C can provide multi-factor authentication with minimal configuration.
+
+Since web APIs have no user interface, they are unable to redirect the user to a secure token service like Azure AD B2C. Instead, the API is passed a bearer token from the calling app, which has already authenticated the user with Azure AD B2C. The API then validates the token without direct user interaction.
+
+In this tutorial, learn how to:
+
+> [!div class="checklist"]
+> * Create an Azure Active Directory B2C tenant
+> * Register a Web API in Azure AD B2C
+> * Use Visual Studio to create a Web API configured to use the Azure AD B2C tenant for authentication
+> * Configure policies controlling the behavior of the Azure AD B2C tenant
+> * Use Postman to simulate a web app which will present a login dialog, retrieve a token, and then use it to make a request against the web API
+
+## Prerequisites
+
+The following are required for this walkthrough:
+
+* [Microsoft Azure subscription](https://azure.microsoft.com/free/?ref=microsoft.com&utm_source=microsoft.com&utm_medium=docs&utm_campaign=visualstudio). 
+* [Visual Studio 2017](https://aka.ms/vsdownload?utm_source=mscom&utm_campaign=msdocs) (any edition)
+* [Postman](https://www.getpostman.com/postman)
+
+## Create the Azure Active Directory B2C tenant
+
+Create an Azure Active Directory B2C tenant [as described in the documentation](/azure/active-directory-b2c/active-directory-b2c-get-started). When prompted, associating the tenant with an Azure subscription is optional for this tutorial.
+
+## Configure a sign-up or sign-in policy
+
+Use the steps in the Azure AD B2C documentation to [create a sign-up or sign-in policy](/azure/active-directory-b2c/active-directory-b2c-reference-policies#create-a-sign-up-or-sign-in-policy). Name the policy **SiUpIn**.  Use the example values provided in the documentation for **Identity providers**, **Sign-up attributes**, and **Application claims**. Using the **Run now** button to test the policy as described in the documentation is optional.
```

Comment:
Remove extra space before the 3rd sentence. 😄 I bet you never saw that comment coming from me.

Reply:
NEVER!  You'll note, I've gotten MUCH better about catching my own...

Reply:
@guardrex That one caught my attention too. I recommend changing the name to something like what you proposed.

Reply:
Funny you guys should mention that.  I had the same thought, but in the end, I decided that since I'm saying, "Hey, go follow the instructions in this link" and those instructions I'm linking to use SiUpIn, I should do the same to be consistent.  Thoughts, @guardrex and @scottaddie ?

Reply:
Can it be changed over there, too?

Reply:
@CamSoper If it's possible to change the name in the instructions you're linking to, I'd highly recommend it. If that's not possible, then stick with "SiUpIn".